### PR TITLE
Add hard-earned new learnings...

### PR DESCRIPTION
**Pull request by MariusVB**

### CODE REVIEW

File: how-we-use-nodejs.md [hunk 1]
```diff
@@ -46,15 +46,16 @@ We use TypeORM as our ORM to integrate with PostgreSQL. We use the repository pa
...
 ```typescript
 // Specify the inverseSide of relation
 @ManyToMany(() => Entity2, (e: Entity2) => e.entity1s)
 // Specify the table name to be used. Always use the same table name for both sides of the relationship
 @JoinTable({ name: 'entity1_to_entity2' })
-// Wrap in Promise, as this relation is lazy-loaded
-entity2s?: Promise<Entity2[]>
+entity2s?: Entity2[]
 
 @ManyToMany(() => Entity1, (e: Entity2) => e.entity2s, { eager: true })
```
Comment (nikolajvonholck) [inline how-we-use-nodejs.md:60]: Why do we have `eager: true`?

File: how-we-use-nodejs.md [hunk 2]
```diff
@@ -46,15 +46,16 @@ We use TypeORM as our ORM to integrate with PostgreSQL. We use the repository pa
...
 @JoinTable({ name: 'entity1_to_entity2' })
```
Comment (nikolajvonholck) [inline how-we-use-nodejs.md:61]: This second `@JoinTable` should be removed.

File: how-we-use-nodejs.md [hunk 3]
```diff
@@ -46,15 +46,16 @@ We use TypeORM as our ORM to integrate with PostgreSQL. We use the repository pa
 
 Read the documentation [here](https://github.com/typeorm/typeorm/blob/master/docs/eager-and-lazy-relations.md)
 
+- If you want to access relations lazily by just writing `entity1.entity2`, you need to wrap the attribute type in `Promise`. However this adds a lot of annoyance when working with the relation, like loading the entity in `__entity2__` if you try to eagerly load it. Therefore it is easier to just load the relation by specifying it in `FindOptions` if you need to access it, like so: `{ relations: ['entity2'] }` or with a `leftJoinAndSelect`if using the `QueryBuilder`
```
Comment (nikolajvonholck) [inline how-we-use-nodejs.md:49]: Could we shorten this to focus on what we do (i.e. `{ relations: ['entity2'] }`) and remove some of what we don't (Promise, eager, etc)?

**Review by nikolajvonholck — CHANGES_REQUESTED** [review]
See my comments.

File: how-we-use-nodejs.md (hunk 2 above)
Comment (nikolajvonholck) [reply how-we-use-nodejs.md:61]: It should only be a decorator on 1 side of the two entities. Consider adding a comment about this :)

File: how-we-use-nodejs.md (hunk 3 above)
Comment (MariusVB) [reply how-we-use-nodejs.md:49]: We do use eager relations as they just "append" `{ relations: ['entity2'] }` to any `find` command on that entity. I have re-written it and focused on not using lazy relations.

File: how-we-use-nodejs.md (hunk 1 above)
Comment (MariusVB) [reply how-we-use-nodejs.md:60]: If we have an entity that should always include a relation, we can specify this. Like an `Attendance` should always have it's `Participant` and `Event` loaded with it.

**Review by nikolajvonholck — APPROVED** [review]
Looks good 👍
